aiocache/decorators.py

```diff
@@ -44,12 +44,38 @@ def __init__(
         self.cache = cache
 
     def __call__(self, f):
-        @functools.wraps(f)
-        async def wrapper(*args, **kwargs):
-            return await self.decorator(f, *args, **kwargs)
+        class CachedFunctionWrapper:
```

Comment:
This should be defined at the top level. We wouldn't be able to reference the type otherwise when we add type annotations.